### ISSUE TITLE
PSScriptAnalyzer Logo or Wallpaper

### ISSUE THREAD

**Issue by bergmeister** [adilio/ps-design#1]
Hi,

your designs look really great. Would you be interested in creating an icon or a Wallpaper for [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer)? There is no business need for it so don't feel obliged to do so but if enjoy doing your work, then I think some folks might also enjoy seeing it in other places. For PSScriptAnalyzer, we could put the logo on the main Readme of the repo or set the wallpaper as the 'social preview' image of the repo to make it more recognizable and stand out.

**Comment by adilio** [comment]
Hey @bergmeister, this sounds like a great idea! I'd love to. Let's aim for both logo and banner/preview. Can you provide a couple ideas for inspiration, and then I can get a draft going?

**Comment by bergmeister** [comment]
Thanks. I don't have anything specific in mind but maybe something similar ESLint for the logo
https://images.app.goo.gl/GUyMvLbxZ9Hfn5QJ7
Or maybe something with PoshChan similar to the GoLang linter?
https://images.app.goo.gl/B8V2y1taLfZkBp7A7

**Comment by wsmelton** [comment]
I'd say a magnifying glass incorporated into the logo would be cool.

https://www.sqlrepairtool.net/img/sql-log-analyzer.png

This is a database tool but just reference to see the idea.

If you have maybe the new PS logo, or modified version of it, with the magnifier on there somewhere.

**Comment by adilio** [comment]
These are both great directions, @wsmelton and @bergmeister! I'll work on some designs with this inspiration , and report back.

**Comment by stevenjudd** [comment]
How about a lightbulb similar to what shows up in VSCode with a green squiggly line as the filament per @jwmoss

**Comment by adilio** [comment]
Hi Peeps!
Sorry I've been a bit preoccupied with career changes, and turning 40!
Will start work on this this week, and should have some designs up by this weekend.
Thanks for the additional suggestion @stevenjudd and @jwmoss !

**Comment by adilio** [comment]
Just put together a draft of @wsmelton's suggestion of a magnifying glass and PowerShell logo. Let me know what y'all think.
![search](https://user-images.githubusercontent.com/21287462/60765586-c4704d00-a051-11e9-9e6f-30fc97626a50.png)

**Comment by bergmeister** [comment]
@adilio Looks really awesome, with lots of detail, I'd be very happy to take it. I like the way the logo is placed inside the magnifier, the non-central position makes it look so much better.
Are the 2 inner bits inside the magnifying glass on the top left shadows? Together with the PS logo, they look a bit confusing, maybe make them smaller or leave them out?

**Comment by adilio** [comment]
He @bergmeister, really enjoying the feedback here and on Twitter here: https://twitter.com/CBergmeister/status/1147794431149518848

I do appreciate Jakub @nohwnd's feedback about the rectangular shape more suited to a logo. I'm going to re-work the design so the magnifying glass fits on the PS Core logo. ALso, I will icorporate the `PSSA` in some way. Stay tuned!

**Comment by adilio** [comment]
Round 2. Incorporating some ideas thus far, including:
- Rectangular (skewed) logo
- PSSA branding
- Green squiggly line

Please provide feedback 😁 .
![pssa2](https://user-images.githubusercontent.com/21287462/60775658-37bda180-a0da-11e9-91d2-8d1e43a2c521.png)

**Comment by compwiz32** [comment]
my suggestion would be to increase the contrast between the magnifying glass and the PS logo behind it. the rim of the magnifying glass can blend into the background in a small version of the logo. there is probably nothing to change with the magnifying glass if the classic blue PS logo is used behind it.

**Comment by adilio** [comment]
Good point, @compwiz32! So, part of my motivation behind lining up the angle of the PS log and magnifying glass handle was to accent the same underlying shape. If that makes things more confusing or unclear, I'm open to tweaking it more. 

I'll continue soliciting suggestions, and roll all feedback into one edit.

**Comment by nohwnd** [comment]
To me the new version is a step back. I really liked the way the magnifying glass inspects the powershell logo. 🙂 Now it does not feel like it anymore, because I can't see the powershell logo, and because I am right-handed the new position of the magnifying glass does not even feel like I am using it. 

Realising what the green squiggly line means also took me way too long. To me it would work if the logo was just "PS Script Analyzer" with the squiggly line under "Analyzer", but in this logo it does not fit.

**Comment by adilio** [comment]
Fair enough, @nohwnd. I guess your suggestion of square logo is what i was going for, and also I'm left-handed so maybe that explains the orientation 🤣. Actually, I did that to make it fit the skew of the PS logo better (as it lines up with angle bracket symbol better).

Your comments are appreciated, though. I don;t want to try and incorporate too much, as the beauty is in the simplicity. Will have to think on this further.

**Comment by johlju** [comment]
Wouldn’t it be enough if it just magnified the logo underneath the magnifying glass, and just skip the letters. :thinking:

**Comment by adilio** [comment]
Hi @bergmeister and all, I've created a folder for this project here:  https://github.com/adilio/ps-design/tree/master/psscriptanalyzer

Based on feedback, I've revised the design to include:
- magnification instead of acronym
- original orientation of magnifying glass, as well as drop shadow to give dimension
- sample of `PSScriptAnalyzer` with green squiggle underneath (this can be used in addition to simple logo, or not)

The evolution continues. Appreciate the feedback.

![pssa3](https://user-images.githubusercontent.com/21287462/60836718-28426500-a17b-11e9-8bfd-3801906f6ef8.png)

**Comment by bergmeister** [comment]
Awesome, this logo is the best of all previous logos, I like the magnifying effect and the asymmetric placements.
I still find the shadows inside the magnifier a bit disturbing. Personally, I am wondering if making the magnifier 10-20% bigger (size still smaller than Ps logo) would make it look even better

**Comment by compwiz32** [comment]
Looks good!

**Comment by stevenjudd** [comment]
This is amazing work. It really looks good.

**Comment by bwright86** [comment]
I am liking the iterations on this. Would it help to put “PSSA” on the magnifying glass handle, and the green squiggly line as well? I think the shadows helps give it depth, but maybe tone them down, so that they are present, but not obvious. Also, just spitballing, but if you wanted to drop the handle and go for a mechanical arm that comes in from the top left or right corner, it could line up with the “>” angle brackets again, or compliment the angles. I am impressed with what you have already done!

**Comment by bwright86** [comment]
Also, in case anyone is curious about the rectangular thingy... it’s called a rhombus, helllooo!


![knights tale gif](https://i.imgur.com/BdxBtsK.gif)

P.S. It took way to long to learn how to put a gif in a comment...

**Comment by adilio** [comment]
The problem with making the shadow any less prominent, and the writing on the handle, is that it is a logo. Logo's get shrunk down real tiny, and you want to be ensure that subtle elements are still visible in smaller form. I have made the shadow a little less subtle, but any more and it will disappear on shrinking.

@bergmeister, based on your suggestion, I've removed the light reflection on magnifying glass lens. Also, I've increased the size of the magnifying glass. Ideally, I'd like to have move the magnifying glass up a little, but I'd have to redo the magnified image, and that's.... effort 🤣.

I do kind of like how the handle hangs down almost into the text.
![pssa4](https://user-images.githubusercontent.com/21287462/60854569-9ad03680-a1b5-11e9-82dd-e92231baa909.png)

**Comment by nohwnd** [comment]
Great work, this got much better proportions and looks like a real logo. 👍

The first version with lens "shadows" looks more refined to me, because it make the magnifying glass look more real. And I don't find it disturbing at all. Especially when I zoom out.
The subtle shadow on the second one is definitely better.

The text needs a bit more work: 
- the shadow on the text is imho unnecessary.
- I would choose another font. Here is a list of some cool free fonts. https://www.logaster.com/blog/best-free-logo-fonts/ Qanelas, beaver, bebas, aileron, maxwell, andis, coves all look interesting. Here a mockup with MuseoSans that I quickly put together in PowerPoint (hence no squiggly line :D) the positioning is far from ideal but the thicker rounder font seems more flattering.

![image](https://user-images.githubusercontent.com/5735905/60876371-1248a900-a23c-11e9-8df7-cd60786ca29a.png)

- The squiggly line should have more height, and maybe repeat slightly more often. Example of squiggly line from vscode. 
![image](https://user-images.githubusercontent.com/5735905/60875419-1a074e00-a23a-11e9-9d6b-cf77be6f1238.png)

**Comment by wsmelton** [comment]
I may have over read the discussion on this but why have text on a logo that won't be readable (e.g. once taken down to a logo size on PS Gallery).

I look at it as once the logo is associated with the module, the text is a mute point. 

Just another point of view on the green line. What about having it outline the inside of the magnifying glass? Just a play on words type thing instead of visually trying to fit a squiggle line (that just like words is not going to be very readable once shrunk down to size).

**Comment by nohwnd** [comment]
@wsmelton I would use that version of the logo with the text and the squiggly line in repo readme, but for the PSGallery I would use just the logo without any text.

**Comment by adilio** [comment]
Appreciate all the feedback.

@nohwnd is right, the logo without text serves a different purpose, whereas the one with text can be used for added recognition (like on stickers).

As we have refined the design a bit, it's coming down to more personal preference, and I'll do my best to cater to whatever the majority decides. But I do think I'm kind of done!

@bergmeister, here is my final-ish version, barring something glaring.

Couple of changes:
- Font to Segoe UI Semi-Bold (MS/Windows font). I prefer to keep it generic. You never know when something is listed free in the internet, and later is not. Copyrights can be pesky.
- Green squiggle given higher frequency, and made taller; also given 60 % opacity
- Text shadow removed
- Magnifying glass reflection brought back (I like it better, too 🤷‍♂)

![pssa5](https://user-images.githubusercontent.com/21287462/60941981-c3c7f880-a295-11e9-9a53-6b37e3d45ba9.png)

And here is just the logo:

![pssa-logo](https://user-images.githubusercontent.com/21287462/60942060-17d2dd00-a296-11e9-988c-34445cd46e85.png)

All versions are in: https://github.com/adilio/ps-design/tree/master/psscriptanalyzer

Have a good night all!

**Comment by nohwnd** [comment]
@adilio looks awesome, great job :)

**Comment by bergmeister** [comment]
Awesome work @adilio I agree the last one is the best of all worlds :-)
I will start various PRs the next day and speak to people. The plan is to add the logo to the repo and then have the `IconUri` point to it so that one sees it on the PSGallery.
There is also a feature on GitHub where one can modify the logo being displayed if one pastes e.g. a link in Twitter. I'll try to also add the logo as a custom emji to the PSSlack channel. Maybe a long shot but files named `PSScriptAnalyzerSettings.psd1`  could display that logo in VSCode as well (the icon repo for it was blocked for a while but got unblocked yesterday). Any other ideas on where to use the icon?

**Comment by adilio** [comment]
Sounds awesome, @bergmeister! I'll have to think about it more and let you know.

I will mention that the SVG file is the vector image, so that's the one you would use to scale up or down. If you need PNG's or other formats in different pixel or size dimensions, please let me know, and I'd be happy to prepare them.

Oh, and if you get a chance, please let the folks in Twitter know about your design and the `ps-design` repo, in case they want to collaborate on a design of their own! 😄 

Cheers!

**Comment by bergmeister** [comment]
Thanks for your hard work, it looks awesome. I will speak to people and try to get things going the next days, updating the PSSA preview picture of the repo on links should be quite quick but it could be that we see the new logo only at the next release in the PSGallery as it's a property in the manifest, I still have to figure that out and play with it. Shall I close the issue now?

**Comment by brwilkinson** [comment]
Awesome job, looks great. . . 

I noticed on the twitter post it came out with a black background . . so the text is hard to see.

![image](https://user-images.githubusercontent.com/3605266/61005028-3085d600-a31c-11e9-8023-767238e00af4.png)

https://twitter.com/adilio/status/1148313230491521024

**Comment by adilio** [comment]
Thanks for the heads up, @brwilkinson!

I have created a "white text" version here:
https://github.com/adilio/ps-design/blob/master/psscriptanalyzer/pssa-final-white-text.png

I have also replied to @bergmeister's thread here:
https://twitter.com/adilio/status/1149063785761521664

Hope that helps!

**Comment by brwilkinson** [comment]
haaa now it looks great on the normal twitter feed and clear on the pop up  

Perhaps a black with a silver outline?  😁

![image](https://user-images.githubusercontent.com/3605266/61009961-6087a600-a329-11e9-87f1-75437d9f7bf1.png)

**Comment by adilio** [comment]
Hehe, this would definitely be a wormhole to go down 🤣.

Looks fine on Twitter with Dark Mode, so I'm just gonna leave it 🤷‍♂. Appreciate the heads up, though!

**Comment by brwilkinson** [comment]
@adilio haa yeah definitely.... 

looks great anyway.
